doc/api/esm.md

```diff
@@ -339,21 +339,23 @@ Node.js and the browser can be written:
 When resolving the `"."` export, if no matching target is found, the `"main"`
 will be used as the final fallback.
 
-The conditions supported in Node.js are matched in the following order:
+The conditions supported in Node.js condition matching:
 
-1. `"node"` - matched for any Node.js environment. Can be a CommonJS or ES
+* `"default"` - the generic fallback that will always match. Can be a CommonJS
+   or ES module file.
+* `"import"` - matched when the package is loaded via `import` or
+   `import()`. Can be any module format, this field does not set the type
+   interpretation. _This is currently only supported behind the
+   `--experimental-conditional-exports` flag._
+* `"node"` - matched for any Node.js environment. Can be a CommonJS or ES
    module file. _This is currently only supported behind the
    `--experimental-conditional-exports` flag._
-2. `"require"` - matched when the package is loaded via `require()`.
+* `"require"` - matched when the package is loaded via `require()`.
    _This is currently only supported behind the
    `--experimental-conditional-exports` flag._
-3. `"import"` - matched when the package is loaded via `import` or
-   `import()`. Can be any module format, this field does not set the type
-   interpretation. _This is currently only supported behind the
-   `--experimental-conditional-exports` flag._
-4. `"default"` - the generic fallback that will always match if no other
-   more specific condition is matched first. Can be a CommonJS or ES module
-   file.
+
+Condition matching is applied in object order from first to last within the
+`"exports"` object.
```

Comment:
nit: can we add a test for duplicated keys?

```json
{
  "node": null,
  "import": ...,
  "node": ...
}
```

mostly just want to ensure we have tests for what object order means here preserved if people are looking at our tests

Reply:
Ok, I've added one. Worth noting that duplicate keys take the value of the last, but the ordering from the first.